aten/src/ATen/native/cpu/MultinomialKernel.cpp

```diff
@@ -55,27 +61,17 @@ void multinomial_apply(Tensor& result, const Tensor& self, const int64_t n_sampl
     TORCH_CHECK(with_replacement || (n_categories - n_zeros >= n_sample),
         "invalid multinomial distribution (with replacement=False, not enough non-negative category to sample)");
 
-    /* normalize cumulative probability distribution so that last val is 1
-    i.e. doesn't assume original self row sums to one */
-    if ((sum > 0) || ((sum < 1.00001) && (sum > 0.99999))) {
-      for (int64_t j = 0; j < n_categories; j++) {
-        cum_dist_ptr[j * cum_dist_stride_0] /= sum;
-      }
-    }
-
     for (int64_t j = 0; j < n_sample; j++) {
       /* sample a probability mass from a uniform distribution */
       at::uniform_real_distribution<double> uniform(0, 1);
-      double uniform_sample = uniform(gen);
+      double uniform_sample = uniform(gen) * sum;
```

Comment:
Why did we stop normalizing the cumulative probability distribution? Does this make it more numerically stable?

Reply:
The normalization would require N additional divisions per round (when with_replacement==False) which are not really necessary...